client/payment-request/redirect-dialog.js

```diff
@@ -0,0 +1,38 @@
+/* global jQuery, wcpayPaymentRequestParams, tb_show */
+jQuery( ( $ ) => {
+	// Remove Thickbox wrapper once it's closed.
+	$( document.body ).on( 'thickbox:removed', () => {
+		$( '.TB_wrapper' ).remove();
+	} );
+} );
+
+export const displayRedirectDialog = () => {
+	tb_show(
+		wcpayPaymentRequestParams.site_url,
+		'#TB_inline?width=400&inlineId=payment-request-redirect-dialog'
+	);
+
+	// Refactor positioning of Thickbox to be mobile-friendly by displaying it
+	// inside a special container.
+	//
+	// For reference, check `tb_show` and `tb_position` in WP core thickbox.js.
+	// Note: We shouldn't change the CSS for the default Thickbox elements,
+	// otherwise it may break the styling in other pages.
+	jQuery( 'body' ).append( '<div class="TB_wrapper"></div>' );
```

Comment:
I'm not familiar but I guess WCPay already loads jQuery so we use jQuery to interact with the DOM by default?